smartsim/ml/tf/utils.py

```diff
@@ -59,7 +61,7 @@ def freeze_model(model: keras.Model, output_dir: str, file_name: str) -> t.Tuple
     if not file_name.endswith(".pb"):
         file_name = file_name + ".pb"
 
-    full_model = tf.function(lambda x: model(x))
+    full_model = tf.function(model)
```

Comment:
I'm not seeing the equivalence here

Reply:
In the lambda, we're using whatever the `__call__` implementation on model is. There should be no difference between passing model and letting that function get executed or passing a lambda that does the same thing, but indirectly. 

Consider:

```
def do_thing(one_arg_func: Callable[[int], None]):
  one_arg_func(42)

f0 = tf.function(lambda x: model(x))
f1 = model

r0 = do_thing(f0)
r1 = do_thing(f1)
```
Both routes pass 42 to the function. f0 pushes 42 down into the lambda to execute `model(42)` but f1 just does `one_arg_fun(42)` (which is `f1(42)` and of course, that's `model(42)` also).

Reply:
Sanity checking by stepping through the `freeze_model` unit test does demonstrate no issues:

![image](https://github.com/CrayLabs/SmartSim/assets/3595025/da56f072-91d3-4570-a582-b17e0b6464fb)

I'm not that familiar with Keras, but [this documentation](https://keras.io/api/models/model/) does talk through creating Model subclasses by implementing the forward pass in `__call__.`